addons/website/controllers/main.py

```diff
@@ -281,7 +285,7 @@ def create_sitemap(url, content):
                 # Sitemaps must be split in several smaller files with a sitemap index
                 content = View._render_template('website.sitemap_index_xml', {
                     'pages': pages_with_website,
-                    'url_root': request.httprequest.url_root,
+                    'url_root': url_root,
```

Comment:
This one is probably wrong, as @JKE-be mentioned.
And this is not what was intended, as I wanted to fix the `<location/>` url mainly

Reply:
@JKE-be I'd like your opinion here.

I am trying to finish / close my old "nice to have" PRs, and here, indeed, a `domain1.com/sitemap-index.xml` can't contain `domain2.com/sitemap-1.xml` entries (different domains). (Note: having `<loc/>` with different domain from the sitemap url is fine.)

We can either do nothing, but even if it's not a real problem (seo/bug), it's still something people keep complaining about a lot. Or we can go with what you had in mind: a cache per domain. It should stop people thinking we are doing something wrong and opening tickets or trying to go with their own fixes.

Is the current diff going in the direction you imagined? It would of course be for master if it was to move forward